src/Meshes/linesMesh.ts

```diff
@@ -102,43 +107,51 @@ export class LinesMesh extends Mesh {
             options.attributes.push(VertexBuffer.ColorKind);
         }
 
-        this._colorShader = new ShaderMaterial("colorShader", this.getScene(), "color", options);
+        this._lineMaterial = new ShaderMaterial("colorShader", this.getScene(), "color", options);
     }
 
     private _addClipPlaneDefine(label: string) {
-        const define = "#define " + label;
-        let index = this._colorShader.options.defines.indexOf(define);
+        if (!this._isShaderMaterial(this._lineMaterial)) {
+            return;
+        }
 
+        const define = "#define " + label;
+        const index = this._lineMaterial.options.defines.indexOf(define);
         if (index !== -1) {
             return;
         }
 
-        this._colorShader.options.defines.push(define);
+        this._lineMaterial.options.defines.push(define);
     }
 
     private _removeClipPlaneDefine(label: string) {
-        const define = "#define " + label;
-        let index = this._colorShader.options.defines.indexOf(define);
+        if (!this._isShaderMaterial(this._lineMaterial)) {
+            return;
+        }
 
+        const define = "#define " + label;
+        const index = this._lineMaterial.options.defines.indexOf(define);
         if (index === -1) {
             return;
         }
 
-        this._colorShader.options.defines.splice(index, 1);
+        this._lineMaterial.options.defines.splice(index, 1);
     }
 
     public isReady() {
         const scene = this.getScene();
 
         // Clip planes
-        scene.clipPlane ? this._addClipPlaneDefine("CLIPPLANE") : this._removeClipPlaneDefine("CLIPPLANE");
-        scene.clipPlane2 ? this._addClipPlaneDefine("CLIPPLANE2") : this._removeClipPlaneDefine("CLIPPLANE2");
-        scene.clipPlane3 ? this._addClipPlaneDefine("CLIPPLANE3") : this._removeClipPlaneDefine("CLIPPLANE3");
-        scene.clipPlane4 ? this._addClipPlaneDefine("CLIPPLANE4") : this._removeClipPlaneDefine("CLIPPLANE4");
-        scene.clipPlane5 ? this._addClipPlaneDefine("CLIPPLANE5") : this._removeClipPlaneDefine("CLIPPLANE5");
-        scene.clipPlane6 ? this._addClipPlaneDefine("CLIPPLANE6") : this._removeClipPlaneDefine("CLIPPLANE6");
-
-        if (!this._colorShader.isReady(this)) {
+        if (!this._isShaderMaterial(this._lineMaterial)) {

```

Comment:
Should be `if (this._isShaderMaterial(this._lineMaterial)) {` instead.

Also, if you keep this check here, you should remove the checks in `_addClipPlaneDefine` and `_removeClipPlaneDefine` as they are redundant.

Reply:
Good catch. Would only prefer checks here but they are added to  `_(add/remove)ClipPlane` due to TS errors. Will remove them if you guys accept code with `//@ts-ignore` (not recommended)

Reply:
Alternatively we can pass material as argument or cast to `ShaderMaterial` in both functions but neither seem elegant.

Reply:
On 2nd thought ....idc! If statement is gone all together from `isReady()` method. Enjoy. 😄